device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsTransport.java

```diff
@@ -439,6 +463,23 @@ public boolean isEmpty()
     }
 
     /**
+     * Registers a callback to be executed whenever the amqps connection is lost or established.
+     * 
+     * @param callback the callback to be called.
+     */
+    public void registerConnectionStateCallback(IotHubConnectionStateCallback callback) {
+        // Codes_SRS_AMQPSTRANSPORT_99_003: [The registerConnectionStateCallback shall register the connection state callback.]
+        this.stateCallbackList.add(callback);
+    }
+
+    /**
+     * Helper method for testing
+     */
+     public int totalConnectionStateCallbacks() {
+         return this.stateCallbackList.size();
```

Comment:
Yes, I will remove this method since it will no longer be needed.  I will also update the spec as well.